### DISCUSSION TITLE
More options around Cookies and Authentication in Axum?

### DISCUSSION THREAD

**Discussion by sidkshatriya** [tokio-rs/axum#296] [Q&A] [answered]
I'm looking for more options around cookies and authentication. Rocket for instance seems to have some options here. Are there some good third party libraries that I can use for authentication and cookies? Things like CSRF protection etc. will be very useful.

**Comment by davidpdrsn** [comment, marked as answer]
axum doesn't have anything built-in for dealing with cookies. We recommend you pick whatever cookies crate fits your need and use that. A few options:

- [headers](https://crates.io/crates/headers). This can be used with the [`TypedHeader`](https://docs.rs/axum/0.2.3/axum/extract/struct.TypedHeader.html) extractor.
- [cookie-rs](https://crates.io/crates/cookie)
- [async-session](https://crates.io/crates/async-session)

See https://github.com/tokio-rs/axum/issues/172 for more details.

**Reply by sidkshatriya** [reply]
(For some reason I'm unable to mark this as the accepted answer. Could be a GitHub frontend bug for my browser. I'll try again later)
